core/src/cluster_info.rs

```diff
@@ -2125,12 +2146,31 @@ impl ClusterInfo {
         feature_set: Option<&FeatureSet>,
         epoch_time_ms: u64,
     ) {
-        let sender = response_sender.clone();
-        thread_pool.install(|| {
-            requests.into_par_iter().for_each_with(sender, |s, reqs| {
-                self.handle_packets(&recycler, &stakes, reqs, s, feature_set, epoch_time_ms)
-            });
+        let mut timer = Measure::start("process_gossip_packets_time");
+        let packets: Vec<_> = thread_pool.install(|| {
+            requests
+                .into_par_iter()
+                .flat_map(|request| request.packets.into_par_iter())
+                .filter_map(|packet| {
+                    let protocol: Protocol =
+                        limited_deserialize(&packet.data[..packet.meta.size]).ok()?;
+                    protocol.sanitize().ok()?;
+                    let protocol = protocol.par_verify()?;
+                    Some((packet.meta.addr(), protocol))
+                })
+                .collect()
         });
+        self.handle_packets(
```

Comment:
Hmmm, so doing the handle_packets in parallel doesn't give benefit?

Reply:
probably it does, but not as much. i will update with more of the packet handling work done on thread pool.